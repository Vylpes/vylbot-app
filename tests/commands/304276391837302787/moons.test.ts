describe("constructor", () => {
    test.todo("EXPECT CommandBuilder to be defined correctly");
});

describe("execute", () => {
    describe("GIVEN interaction is not a chat input command", () => {
        test.todo("EXPECT nothing to happen");
    });

    describe("GIVEN interaction subcommand is list", () => {
        test.todo("EXPECT ListMoons to be called");
    });

    describe("GIVEN interaction subcommand is add", () => {
        test.todo("EXPECT AddMoon to be called");
    });
});