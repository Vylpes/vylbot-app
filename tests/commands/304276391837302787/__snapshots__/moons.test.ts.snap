// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constructor EXPECT CommandBuilder to be defined correctly 1`] = `
{
  "contexts": undefined,
  "default_member_permissions": undefined,
  "default_permission": undefined,
  "description": "View and create moons",
  "description_localizations": undefined,
  "dm_permission": undefined,
  "integration_types": undefined,
  "name": "moons",
  "name_localizations": undefined,
  "nsfw": undefined,
  "options": [
    {
      "description": "List moons you have obtained",
      "description_localizations": undefined,
      "name": "list",
      "name_localizations": undefined,
      "options": [
        {
          "description": "The user to view (Defaults to yourself)",
          "description_localizations": undefined,
          "name": "user",
          "name_localizations": undefined,
          "required": false,
          "type": 6,
        },
        {
          "autocomplete": undefined,
          "choices": undefined,
          "description": "The page to start with",
          "description_localizations": undefined,
          "max_value": undefined,
          "min_value": undefined,
          "name": "page",
          "name_localizations": undefined,
          "required": false,
          "type": 10,
        },
      ],
      "type": 1,
    },
    {
      "description": "Add a moon to your count!",
      "description_localizations": undefined,
      "name": "add",
      "name_localizations": undefined,
      "options": [
        {
          "autocomplete": undefined,
          "choices": undefined,
          "description": "What deserved a moon?",
          "description_localizations": undefined,
          "max_length": undefined,
          "min_length": undefined,
          "name": "description",
          "name_localizations": undefined,
          "required": true,
          "type": 3,
        },
      ],
      "type": 1,
    },
  ],
  "type": 1,
}
`;
